File: twoSum.js
function twoSum(nums, target) {
    let map = new Map(); // Create a new map to store numbers and their indices
    for (let i = 0; i < nums.length; i++) { // Loop through the array
        let complement = target - nums[i]; // Calculate the complement (target - current number)
        if (map.has(complement)) { // Check if the complement exists in the map
            return [map.get(complement), i]; // If yes, return its index and current index
        }
        map.set(nums[i], i); // Otherwise, store the current number and its index in the map
    }
    return []; // If no solution found, return an empty array
}

// Example input
const nums = [6, 5, 3, 2];
const target = 11;

// Running the function
const result = twoSum(nums, target);
console.log(result);  // Output: [1, 2]

File: twoSum.cpp
#include <iostream>
#include <unordered_map>
#include <vector>

using namespace std;

vector<int> twoSum(vector<int>& nums, int target) {
    unordered_map<int, int> map;
    for (int i = 0; i < nums.size(); i++) {
        int complement = target - nums[i];
        if (map.find(complement) != map.end()) {
            return {map[complement], i};
        }
        map[nums[i]] = i;
    }
    return {}; // If no solution found
}

int main() {
    vector<int> nums = {6, 5, 3, 2};
    int target = 11;

    vector<int> result = twoSum(nums, target);

    if (!result.empty()) {
        cout << "Indices: [" << result[0] << ", " << result[1] << "]" << endl;
    } else {
        cout << "No solution found" << endl;
    }

    return 0;
}
